starfish/decoders/iss.py

```diff
@@ -2,44 +2,54 @@
 import pandas as pd
 
 
-def decode(spots_df_tidy, letters=['T', 'G', 'C', 'A']):
-    num_ch = spots_df_tidy.ch.max() + 1
-    num_hy = spots_df_tidy.hyb.max() + 1
-    num_spots = spots_df_tidy.spot_id.max() + 1
+class IssDecoder:
+    def __init__(self, encoded, codebook):
```

Comment:
API-wise, I think it makes more sense for the decoder to be instantiated with the codebook, and the decode method to take the encoded data as an input.

Reply:
👍